internal/comparetest/compare.go

```diff
@@ -267,3 +268,210 @@ func numberTypeToString(t pmetric.NumberDataPointValueType) string {
+
+	var errs error
+	for e := 0; e < numLogRecords; e++ {
+		elr := expected.At(e)
+		var foundMatch bool
+		for a := 0; a < numLogRecords; a++ {
+			alr := actual.At(a)
+			if _, ok := matchingLogRecords[alr]; ok {
+				continue
+			}
+			if reflect.DeepEqual(elr.Attributes().AsRaw(), alr.Attributes().AsRaw()) {
+				foundMatch = true
+				matchingLogRecords[alr] = elr
+				break
+			}
+		}
+		if !foundMatch {
+			errs = multierr.Append(errs, fmt.Errorf("log missing expected resource with attributes: %v", elr.Attributes().AsRaw()))
+		}
+	}
+
+	for i := 0; i < numLogRecords; i++ {
+		if _, ok := matchingLogRecords[actual.At(i)]; !ok {
+			errs = multierr.Append(errs, fmt.Errorf("log has extra record with attributes: %v", actual.At(i).Attributes().AsRaw()))
+		}
+	}
+
+	if errs != nil {
+		return errs
+	}
+
+	for alr, elr := range matchingLogRecords {
+		if err := CompareLogRecords(alr, elr); err != nil {
+			return multierr.Combine(fmt.Errorf("log record with attributes: %v, does not match expected", alr.Attributes().AsRaw()), err)
+		}
+	}
+	return nil
+}
+
+// CompareLogRecords compares each part of two given LogRecord and returns
+// an error if they don't match. The error describes what didn't match.
+func CompareLogRecords(expected, actual plog.LogRecord) error {
+	if expected.Flags() != actual.Flags() {
+		return fmt.Errorf("log record Flags doesn't match expected: %d, actual: %d",
+			expected.Flags(),
+			actual.Flags())
+	}
+
+	if expected.DroppedAttributesCount() != actual.DroppedAttributesCount() {
+		return fmt.Errorf("log record DroppedAttributesCount doesn't match expected: %d, actual: %d",
+			expected.DroppedAttributesCount(),
+			actual.DroppedAttributesCount())
+	}
+
+	if expected.Timestamp() != actual.Timestamp() {
+		return fmt.Errorf("log record Timestamp doesn't match expected: %d, actual: %d",
+			expected.Timestamp(),
+			actual.Timestamp())
+	}
+
+	if expected.ObservedTimestamp() != actual.ObservedTimestamp() {
+		return fmt.Errorf("log record ObservedTimestamp doesn't match expected: %d, actual: %d",
+			expected.ObservedTimestamp(),
+			actual.ObservedTimestamp())
+	}
+
+	if expected.SeverityNumber() != actual.SeverityNumber() {
+		return fmt.Errorf("log record SeverityNumber doesn't match expected: %d, actual: %d",
+			expected.SeverityNumber(),
+			actual.SeverityNumber())
+	}
+
+	if expected.SeverityText() != actual.SeverityText() {
+		return fmt.Errorf("log record SeverityText doesn't match expected: %s, actual: %s",
+			expected.SeverityText(),
+			actual.SeverityText())
+	}
+
+	if expected.TraceID() != actual.TraceID() {
+		return fmt.Errorf("log record TraceID doesn't match expected: %d, actual: %d",
+			expected.TraceID(),
+			actual.TraceID())
+	}
+
+	if expected.SpanID() != actual.SpanID() {
+		return fmt.Errorf("log record SpanID doesn't match expected: %d, actual: %d",
+			expected.SpanID(),
+			actual.SpanID())
+	}
+
+	if !expected.Body().Equal(actual.Body()) {
```

Comment:
Map.Equal method is deprecated

```suggestion
	if expected.Body().AsRaw() != actual.Body().AsRaw() {
```